### PR TITLE
Enable artifactory publication

### PR DESCRIPTION
**Pull request by samtstern**
This enables publication of `-SNAPSHOT` builds on jFrog artifactory.  Once I get it configured, this can run from Travis (!!!)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -206,16 +215,27 @@ allprojects { project ->
             }
         }
 
+        artifactory {
+            contextUrl = 'http://localhost:8081/artifactory'
```
Comment (samtstern) [inline build.gradle]: Need to replace this with real credentials for oss.jfrog

File: scripts/artifactory.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+
+./gradlew :library:prepareArtifacts
```
Comment (samtstern) [inline scripts/artifactory.sh]: I can't explain it, but this script is the only way I can get things to work. Gradle task dependencies randomly skips some of these.

**Comment by samtstern** [comment]
I must have messed something up because I am getting a ton of lint errors like this:

```
... can only be called from within the same library group (groupId=artifacts)
```

**Comment by samtstern** [comment]
Woo!
https://oss.jfrog.org/webapp/#/artifacts/browse/tree/PomView/oss-snapshot-local/com/firebaseui/firebase-ui-database/3.2.2-SNAPSHOT/firebase-ui-database-3.2.2-20180213.161706-1.pom

File: scripts/artifactory.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+set -e
```
Comment (SUPERCILEX) [inline scripts/artifactory.sh]: I believe this can be inlined: `#!/usr/bin/env bash -e`.

File: gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -2,4 +2,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.5-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.5-bin.zip
```
Comment (SUPERCILEX) [inline gradle/wrapper/gradle-wrapper.properties]: Whoops, looks like this was a mistake.

File: scripts/artifactory.sh (hunk 3 above)
Comment (samtstern) [reply scripts/artifactory.sh]: It didn't like that :man_shrugging:

File: gradle/wrapper/gradle-wrapper.properties (hunk 4 above)
Comment (samtstern) [reply gradle/wrapper/gradle-wrapper.properties]: :taco:

**Pull request merged**